desktop/app/kbfs-helper.js

```diff
@@ -1,10 +1,32 @@
 // @flow
 
 import {ipcMain, shell} from 'electron'
+import fs from 'fs'
+import {pathToURL} from './paths'
 
 export default function () {
   ipcMain.on('openInKBFS', (e, path) => {
-    // We do the `shell.openItem` here to not block the ui thread
-    shell.openItem(path)
+    fs.stat(path, (err, stats) => {
```

Comment:
You ca profile the main thread to understand what's happening better. See
electrons docs. We have helpers to do this in our package.json

On Tuesday, August 2, 2016, Alex Wendland <notifications@github.com> wrote:

> In desktop/app/kbfs-helper.js
> <https://github.com/keybase/client/pull/3718#discussion_r73261349>:
>
> >
> >  export default function () {
> >    ipcMain.on('openInKBFS', (e, path) => {
> > -    // We do the `shell.openItem` here to not block the ui thread
> > -    shell.openItem(path)
> > +    fs.stat(path, (err, stats) => {
>
> It may actually be a subsequent call that does the hanging, I'm not sure.
> But one of these fs calls hangs the entire app.
>
> —
> You are receiving this because you are on a team that was mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/keybase/client/pull/3718/files/e41ff3558a85750b165475ff083fed2d184c7ece#r73261349>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAfhUyQvNTNOHPrb7k5YtNjtSAoO8b15ks5qb9yfgaJpZM4JbIxN>
> .
>